config.yaml

```diff
@@ -0,0 +1,36 @@
+openai:
```

Comment:
That is a nice config implementation. I like using pydantic-settings and use inheritance to create this type of config but from env vars to avoid code changes (or repo changes triggering the CICD pipeline) and just do it at a infrastructure level. But it is just a personal preference
https://docs.pydantic.dev/latest/concepts/pydantic_settings/#usage